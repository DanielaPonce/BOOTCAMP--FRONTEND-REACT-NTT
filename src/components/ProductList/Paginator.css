.paginator-container {
	display: flex;
	align-items: center;
	column-gap: 16px;
}

.paginator-container button {
	box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
	background-color: var(--white);
	border-radius: 50%;
    border: unset;
	padding: 4px 8px;
	cursor: pointer;
    transition: all 0.3s;
}

.paginator-container button:hover {
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
}