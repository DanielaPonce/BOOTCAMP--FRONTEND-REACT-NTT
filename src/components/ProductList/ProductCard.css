
.product-card {
	background-color: var(--white);
	border-radius: 10px;
	box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
	overflow: hidden;
	width: 300px;
	transition: transform 0.3s;

	position: relative;
}

.product-card:hover {
	transform: translateY(-16px);
}

.product-card img {
	width: 100%;
	height: 200px;
	object-fit: contain;
}

.product-details {
	padding: 1rem;
	text-align: center;
}

.product-details h3 {
	margin: 0.5rem 0;
}

.product-details p {
	font-size: 0.9rem;
	color: var(--text-color);
}

.product-details .category {
	font-weight: bold;
	color: var(--secondary-color);
	background-color: var(--white);
	padding: 4px 7px;
	border-radius: 4px;
	display: inline-block;
	margin-bottom: 0.5rem;

	position: absolute;
	z-index: 1;
	top: 2%;
	left: 2%;
}

.product-details .price {
	font-size: 1.2rem;
	color: var(--text-color);
	margin: 0.5rem 0;
}

.product-details button {
	background-color: var(--secondary-color);
	color: var(--white);
	padding: 0.7rem 1.5rem;
	border: none;
	border-radius: 5px;
	cursor: pointer;
	transition: background-color 0.3s;
}

.product-details button:hover {
	background-color: var(--hover-button-color);
}
