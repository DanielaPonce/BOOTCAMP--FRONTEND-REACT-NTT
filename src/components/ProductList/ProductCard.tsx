import { FC } from 'react';
import { Product } from '../../models/product.types';
import { useCart } from '../../hooks/useCart';

const ProductCard: FC<Product> = (props) => {
	const { id, title, description, price, thumbnail, category } = props;

	const { addToCart } = useCart();

	const handleAddToCart = () => {
		addToCart(props);
	};

	return (
		<div className="product-card">
			<img src={thumbnail} alt={title} />
			<div className="product-details">
				<h3>{title}</h3>
				<p>{description}</p>
				<span className="category">{category}</span>
				<div className="price">Precio: S/ {price}</div>
				<button onClick={handleAddToCart}>Agregar al carrito</button>
			</div>
		</div>
	);
};

export default ProductCard;
