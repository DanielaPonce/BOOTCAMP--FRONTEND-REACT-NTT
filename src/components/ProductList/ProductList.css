.product-container {
	display: grid;
	grid-template-columns: repeat(4, 1fr);
	justify-content: center;
	gap: 3rem 1.5rem;
}

@media only screen and (min-width: 992px) and (max-width: 1439px) {
	.product-container {
		grid-template-columns: 1fr 1fr 1fr;
	}
}

@media only screen and (min-width: 740px) and (max-width: 991px) {
	.product-container {
		grid-template-columns: 1fr 1fr;
	}
}

@media only screen and (max-width: 739px) {
	.product-container {
		grid-template-columns: 1fr;
		gap: 16px;
	}
}