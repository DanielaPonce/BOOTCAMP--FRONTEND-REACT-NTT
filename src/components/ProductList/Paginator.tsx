import './Paginator.css';

type PaginatorProps = {
	currentPage: number;
	totalPages: number;
	onPageChange: (page: number) => void;
};

const Paginator: React.FC<PaginatorProps> = ({
	currentPage,
	totalPages,
	onPageChange
}) => {
	return (
		<div className="paginator-container">
			<button
				onClick={() => onPageChange(currentPage - 1)}
				disabled={currentPage === 1}
			>
				&lt;
			</button>
			<div id="paginator-content__number">
				Page {currentPage} of {totalPages}
			</div>
			<button
				onClick={() => onPageChange(currentPage + 1)}
				disabled={currentPage === totalPages}
			>
				&gt;
			</button>
		</div>
	);
};

export default Paginator;
