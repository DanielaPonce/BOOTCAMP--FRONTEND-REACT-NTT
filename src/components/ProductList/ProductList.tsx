import { FC } from 'react';
import { Product } from '../../models/product.types';
import ProductCard from './ProductCard';

type ProductListProps = {
	products: Product[];
};

const ProductList: FC<ProductListProps> = ({ products }) => {
	return (
		<div id="productContainer" className="product-container">
			{products.map((product) => (
				<ProductCard key={product.id} {...product} />
			))}
		</div>
	);
};

export default ProductList;
