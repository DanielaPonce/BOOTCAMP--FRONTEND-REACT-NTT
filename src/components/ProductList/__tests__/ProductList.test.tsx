import React from 'react';
import { render, screen } from '@testing-library/react';
import ProductList from '../ProductList';
import ProductCard from '../ProductCard';
import { Product } from '../../../models/product.types';

jest.mock('../ProductCard', () =>
	jest.fn(() => <div data-testid="product-card"></div>)
);

describe('ProductList Component', () => {
	const mockProducts: Product[] = [
		{
			id: 1,
			title: 'Producto 1',
			description: 'Descripción 1',
			price: 10.0,
			thumbnail: 'https://via.placeholder.com/150',
			category: 'Categoría 1'
		},
		{
			id: 2,
			title: 'Producto 2',
			description: 'Descripción 2',
			price: 20.0,
			thumbnail: 'https://via.placeholder.com/150',
			category: 'Categoría 2'
		}
	];

	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('renders the correct number of ProductCard components', () => {
		render(<ProductList products={mockProducts} />);

		const productCards = screen.getAllByTestId('product-card');
		expect(productCards).toHaveLength(mockProducts.length);
	});

	test('renders no ProductCard components when product list is empty', () => {
		render(<ProductList products={[]} />);

		const productCards = screen.queryByTestId('product-card');
		expect(productCards).toBeNull();
	});

	test('passes correct props to ProductCard', () => {
		render(<ProductList products={mockProducts} />);

		mockProducts.forEach((product, index) => {
			expect(ProductCard).toHaveBeenNthCalledWith(
				index + 1,
				{ ...product },
				{}
			);
		});
	});

	test('matches snapshot', () => {
		const { asFragment } = render(<ProductList products={mockProducts} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
