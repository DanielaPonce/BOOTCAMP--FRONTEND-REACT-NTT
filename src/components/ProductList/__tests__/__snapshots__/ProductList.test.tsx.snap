// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="product-container"
    id="productContainer"
  >
    <div
      data-testid="product-card"
    />
    <div
      data-testid="product-card"
    />
  </div>
</DocumentFragment>
`;
